### PR TITLE
Fixes #8246: Node configuration cache is invalidated too often

### PR DESCRIPTION
**Pull request by fanf**
https://www.rudder-project.org/redmine/issues/8246

### CODE REVIEW

File: rudder-core/src/main/scala/com/normation/rudder/services/policies/DeploymentService.scala [hunk 1]
```diff
@@ -439,10 +439,13 @@ trait DeploymentService_buildNodeConfigurations extends DeploymentService with L
     , globalSystemVariables: Map[String, Variable]
   ): Map[NodeId, InterpolationContext] = {
 
+    println(s"**** AGAIN ${allNodeInfos.size} nodes in allNodeInfos, check exists:\n *** ${List("06da3556-5204-4bd7-b3b0-fa5e7bcfbbea","009e0af1-1a3c-4ee0-a91d-10b68a008f5d", "073239d7-c633-4b60-8477-57131839fd34").map( k => (k, allNodeInfos.isDefinedAt(NodeId(k)))).mkString("\n *** ")}")
```
Comment (matya) [inline rudder-core/src/main/scala/com/normation/rudder/services/policies/DeploymentService.scala]: What is this line doing :o
Are the hard-coded nodeIds not just for debug?

Comment (VinceMacBuche) [inline rudder-core/src/main/scala/com/normation/rudder/services/policies/DeploymentService.scala]: drunk @fanf is drunk ;) I think it's debug / dev logger that need to be removed ... or changed as logger.trace without hardcoded ids ...

**Comment by VinceMacBuche** [comment]
apart the print it's ok to me and can be merged !

File: rudder-core/src/main/scala/com/normation/rudder/services/policies/DeploymentService.scala (hunk 1 above)
Comment (matya) [inline rudder-core/src/main/scala/com/normation/rudder/services/policies/DeploymentService.scala]: Pay attention to the Balmer Peak!
https://xkcd.com/323/

Comment (fanf) [inline rudder-core/src/main/scala/com/normation/rudder/services/policies/DeploymentService.scala]: It was debug. The nodes in the list had specific interesting properties, like "no policy server", "policy server id not a known node", "policy server a known node (which don't know it is a policy server)"

**Comment by fanf** [comment]
PR updated

**Pull request merged**